imagepipeline/src/main/java/com/facebook/imagepipeline/cache/BufferedDiskCache.java

```diff
@@ -187,6 +187,11 @@ public EncodedImage call()
                   try {
                     result = new EncodedImage(ref);
```

Comment:
This line would not throw an `Exception` for corrupted data. The `EncodedImage` is just a wrapper around the encoded image bytes/inputstream.

It is the `DecodeProducer` who would fail during the pipeline processing. The path triggered by bad images is `DecodeProducer#handleError`. I think one possibility would to propagate the failure to the `BaseProducerContext`. Then you are on a higher level where you can find a way to issue a remove to the caches.